### DISCUSSION TITLE
`Url parameter not allowed` in production on firebase

### DISCUSSION THREAD

**Discussion by AnuragRamdasan** [vercel/next.js#19389] [Help]
Hi,

Been working on a new project on firebase for a while now. For this I am keeping my images on s3 while the site is deployed on firebase. my current next.config.js looks like this

```javascript
const withImages = require("next-images");
const withPlugins = require("next-compose-plugins");

const nextConfig = {
  distDir: "../.next",
  images: {
    domains: [
      "project-attachments.s3.us-west-1.amazonaws.com",
      "project-blogs.s3-us-west-1.amazonaws.com",
    ],
  },
};

module.exports = withPlugins([[withImages]], nextConfig);
```

The url for my images come from a different graphql server and is passed down to the components and is eventually rendered by `next/image` by passing the S3 image url as `src`.

This works fine for me on `yarn dev` as well as `yarn build && yarn start`.

However as soon as I push this to firebase, this seems to break for all images.
The image requests error out with a 400 error code with the message `"url" parameter is not allowed`

Images bundled with the codebase work perfectly fine which leads me to think this is an issue with however next.config.js is being read on firebase. Would love to know if this is a known issue.

**Comment by CervarlCG** [comment]
Hi, i have the same issue, do you found any solution?

**Reply by AnuragRamdasan** [reply]
removed it and moved entirely to Imgix. Unfortunately there was no other way to make it work.

**Reply by CervarlCG** [reply]
I solved it.

In my functions file of firebase i was overwriting the nextjs configuration so i only import the nextjs.config.js and passed it to next config

```
const config = require('next.config.js')
const app = next({
    config
});
```

And it works

**Reply by maurocolella** [reply]
That got me one step further, but unfortunately, I am getting `500 Internal Server Error`.

**Comment by kaylabranch** [comment]
@maurocolella What does your next.config.js look like? My images were working fine locally but in prod I got the same error you saw. I had to make sure my domains were included in the images object, looks like this:

```
module.exports = {
  images: {
    domains: ['localhost', 'mydomain.dev', 'cms.mydomain.dev']
  },
  ...
}
```

**Comment by maurocolella** [comment]
My domains are included. I simply get an internal server 500 error on the
next image script, and don't currently have time to track the underlying
stack trace.

**Comment by yourkingnico** [comment]
Incase anyone else is stuck with the _"url" parameter is not allowed_ message, I was able to get past it locally by adding domains in my next.config.js:
`module.exports = {
  images: {
    remotePatterns: [
      {
       ...
      },
    ],
    domains: [
      'localhost'
    ]
  },
}`
